src/battle_ai_main.c

```diff
@@ -3592,7 +3592,7 @@ static u32 AI_CalcMoveScore(u32 battlerAtk, u32 battlerDef, u32 move)
             else
                 ADJUST_SCORE(WEAK_EFFECT);
         }
-        else if (aiData->abilities[battlerAtk] != ABILITY_CONTRARY)
+        else
```

Comment:
This check was unnecessary as `IncreaseStatUpScore` already checks for contrary.